### PR TITLE
Fix: drift in value of env0_configuration_variable in every plan

### LINKED ISSUE #935: Drift in `value` of `env0_configuration_variable` in every plan
**Describe the bug**
Defining a `env0_configuration_variable` resource having `is_sensitive = true` determines the provider to think the value has changed on every plan.

**To Reproduce**

```hcl
variable "test_secret" {
  type = string
}

resource "env0_configuration_variable" "secret" {
  name         = "this_is_a_test_secret"
  value        = var.test_secret
  is_sensitive = true
  type         = "terraform"
  project_id   = "<MY_PROJECT_ID>"
}
```

1. `terraform init`
2. `terraform apply -var=test_secret=1234`
3. `terraform plan -var=test_secret=1234` -> You see the following:

```
# env0_configuration_variable.secret will be updated in-place
  ~ resource "env0_configuration_variable" "secret" {
        id           = "<MY_ID>"
        name         = "this_is_a_test_secret"
      ~ value        = (sensitive value)
        # (6 unchanged attributes hidden)
    }
```

**Expected behavior**
No drift should happen when planning a non-changed configuration.

**Provider Version**
1.20.1

**Screenshots**
No screenshots

**Additional context**
No additional context

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
fixes #935 

### Solution

1. Updated struct to avoid using automatic schema handling for value.
2. Value is now in "read operations" only if it's not senstive. Fixed in both resource and data.
3. Fixed env0_source_code_variables as well.
4. Updated acceptance tests.

### CODE REVIEW

File: client/configuration_variable.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func (c *ConfigurationVariableSchema) ResourceDataSliceStructValueWrite(values m
 
 type ConfigurationVariable struct {
 	ScopeId        string                       `json:"scopeId,omitempty"`
-	Value          string                       `json:"value"`
+	Value          string                       `json:"value" tfschema:"-"`
```
Comment (TomerHeber) [inline client/configuration_variable.go:44]: this avoid writing the value to the schema. Special handling is required instead.

File: env0/resource_configuration_variable_test.go [hunk 2]
```diff
@@ -74,6 +76,47 @@ func TestUnitConfigurationVariableResource(t *testing.T) {
 		})
 	})
 
+	t.Run("Create sensitive", func(t *testing.T) {
```
Comment (TomerHeber) [inline env0/resource_configuration_variable_test.go:79]: this tests the fix for the bug.

File: env0/data_source_code_variables.go [hunk 3]
```diff
@@ -76,10 +76,20 @@ func dataSourceCodeVariablesRead(ctx context.Context, d *schema.ResourceData, me
...
 	}
 
-	if err := writeResourceDataSlice(variables, "variables", d); err != nil {
+	ivalues, err := writeResourceDataGetSliceValues(variables, "variables", d)
+	if err != nil {
 		return diag.Errorf("schema slice resource data serialization failed: %v", err)
 	}
 
+	for i, ivalue := range ivalues {
+		if variables[i].IsSensitive == nil || !*variables[i].IsSensitive {
```
Comment (TomerHeber) [inline env0/data_source_code_variables.go:85]: Also requires special handling.

File: env0/resource_configuration_variable.go [hunk 4]
```diff
@@ -223,13 +225,17 @@ func resourceConfigurationVariableRead(ctx context.Context, d *schema.ResourceDa
 		return diag.Errorf("schema resource data serialization failed: %v", err)
 	}
 
+	if variable.IsSensitive == nil || !*variable.IsSensitive {
```
Comment (TomerHeber) [inline env0/resource_configuration_variable.go:228]: special handling...

File: env0/data_configuration_variable.go [hunk 5]
```diff
@@ -136,6 +136,10 @@ func dataConfigurationVariableRead(ctx context.Context, d *schema.ResourceData,
 		return diag.Errorf("schema resource data serialization failed: %v", err)
 	}
 
+	if variable.IsSensitive == nil || !*variable.IsSensitive {
```
Comment (TomerHeber) [inline env0/data_configuration_variable.go:139]: special handling...

**Review by yaronya — APPROVED** [review]

**Pull request merged**